openwisp_monitoring/device/migrations/__init__.py

```diff
@@ -1,46 +0,0 @@
-    },
-)
-TEMPLATE_CRONTAB_MONITORING_01 = OrderedDict(
-    {
-        "path": "/etc/crontabs/root",
-        "mode": "0644",
-        "contents": "*/5 * * * * /usr/sbin/openwisp-monitoring\n",
-    }
-)
-TEMPLATE_RC_LOCAL_01 = OrderedDict(
-    {
-        "path": "/etc/rc.local",
-        "mode": "0644",
-        "contents": "# Put your custom commands here that should be executed once\n# the system init finished. By default this file does nothing.\n\n/usr/sbin/openwisp-monitoring\ntouch /etc/crontabs/root\n/etc/init.d/cron start\n\nexit 0\n",  # noqa
-    }
-)
-TEMPLATE_UPDATE_OPENWISP_PACKAGES_01 = OrderedDict(
-    {
-        "path": "/usr/sbin/update-openwisp-packages",
-        "mode": "0744",
-        "contents": "#!/bin/sh\nreboot=false\n\n# updates opkg lists only if necessary\nopkg_update(){\n    (\n        test -d /tmp/opkg-lists/ && \\\n        test -f /tmp/opkg-lists/openwrt_base && \\\n        test -f /tmp/opkg-lists/openwrt_packages && \\\n        test -f /tmp/opkg-lists/openwrt_core\n    ) || opkg update;\n}\n\n# installs libubus-lua if necessary\nlibubus_lua_installed=$(opkg list-installed | grep libubus-lua -c)\nif [ \"$libubus_lua_installed\" == \"0\" ]; then\n    opkg_update\n    opkg install libubus-lua\nfi\n\n# installs lua-cjson if necessary\nluacjson_installed=$(opkg list-installed | grep lua-cjson -c)\nif [ \"$luacjson_installed\" == \"0\" ]; then\n    opkg_update\n    opkg install lua-cjson\nfi\n\n# installs rpcd-mod-iwinfo if necessary\nrpcd_mod_iwinfo_installed=$(opkg list-installed | grep rpcd-mod-iwinfo -c)\nif [ \"$rpcd_mod_iwinfo_installed\" == \"0\" ]; then\n    opkg_update\n    opkg install rpcd-mod-iwinfo\n    reboot=true \nfi\n\n# upgrades openwisp-config if necessary\nopenwisp_config_version=$(openwisp_config --version)\nif [ \"$openwisp_config_version\" != \"openwisp-config 0.5.0\" ]; then\n    # backup config just in case...\n    cp /etc/config/openwisp /etc/config/openwisp-backup\n    opkg_update\n    opkg install http://downloads.openwisp.io/openwisp-config/2021-01-07-162007/openwisp-config-mbedtls_0.5.0-1_all.ipk\n    # restore backup\n    mv /etc/config/openwisp-backup /etc/config/openwisp\n    # remove default conf\n    rm /etc/config/openwisp-opkg\n    /etc/init.d/openwisp_config restart\nfi\n\n# reboots if rpcd-mod-iwinfo has been installed\nif [ \"$reboot\" == \"true\" ]; then\n    sleep 5\n    reboot && exit\nfi\n",  # noqa
-    }
-)
-TEMPLATE_POST_RELOAD_HOOK_01 = OrderedDict(
-    {
-        "path": "/etc/openwisp/post-reload-hook",
-        "mode": "0744",
-        "contents": "#!/bin/sh\ntouch /etc/crontabs/root\n/etc/init.d/cron start\n/usr/sbin/update-openwisp-packages\n/usr/sbin/openwisp-monitoring\n",  # noqa
-    }
-)
```

Comment:
as said, I would not remove this code but add a new migration which introduces the following changes:

- ensure the name of the files do not collide with the openwrt openwisp monitoring package
- change the code of the monitoring template so that if it detects the openwrt monitoring package agent it stops and doesn't collect/send data
- remove the automatic install of openwisp-config since it's painful to maintain and can cause issues

Reply:
@nemesisdesign I have updated the `/usr/sbin/update-openwisp-packages"` script and only removed the part that installed openwisp-config.